docs/debugging.md

```diff
@@ -0,0 +1,101 @@
+## Debugging
+
++ [Components](#components)
++ [Actions](#actions)
++ [Reducers](#reducers)
++ [Client](#client)
++ [Communication](#communication)
+
+The best thing about React and Redux is that it simplifies the development workflow.
+Generally speaking, code belongs in one of four buckets: component, action, reducer, or client.
+
+There are different strategies for debugging each bucket, which we'll outline below.
+
+### Components
+
+The first thing you want to find the component to work on.
+The react devtools extension is useful for find the correct component.
+
+Once you find the component, check the component's state and props and monitor the render calls.
+The easiest way to do this is to add a breakpoint in the render function.
+
+### Actions
+
+Actions are the Debugger's API for adding breakpoints, stepping, and everything else.
+
+Sometimes you want to know how to invoke an action or figure out why it's not working.
+Sometimes you want to understand why the state is what it is and what actions have occurred.
+
+A good place to start is logging. You can log the Debugger's actions by enabling `logging.actions` in your local config.
+
+Once you've narrowed the question down to an action you can debug it either in the console or unit tests.
+
+It's easy to try invoking an action in the console with the `getGlobalsForTesting` helper:
+
+```js
+getGlobalsForTesting().actions.selectSource()
+```
+
+We have unit tests for several of our actions in `src/actions/tests`.
+It's nice to look at the unit tests to see how an action *should* be called. If you don't see the use case you're looking for, add a unit test to try it out. Once you get it working you can PR the new unit test and everyone benefits :)
+
+### Reducers
+
+It's really common to want to know what data the Debugger has.
+For instance, what does a breakpoint look like?
+In Redux, the reducers are like the application database and it's tremendously useful to be able to inspect them.
+
+It's easy to view the current Debugger state in the console with
+the `appStore` global. Because the data is immutable, you'll need to request the state as a pojo.
```

Comment:
don't know if we could have `<abbr>` in markdown, but here `<abbr title="Plain Old Javascript Object">pojo</abbr>` would be nice